client/src/extension.ts

```diff
@@ -6,193 +6,24 @@
-    // Register the server for Angular templates and TypeScript documents
-    documentSelector: [
-      // scheme: 'file' means listen to changes to files on disk only
-      // other option is 'untitled', for buffer in the editor (like a new doc)
-      {scheme: 'file', language: 'html'},
-      {scheme: 'file', language: 'typescript'},
-    ],
-
-    synchronize: {
-      fileEvents: [
-        // Notify the server about file changes to tsconfig.json contained in the workspace
-        vscode.workspace.createFileSystemWatcher('**/tsconfig.json'),
-      ]
-    },
-
-    // Don't let our output console pop open
-    revealOutputChannelOn: lsp.RevealOutputChannelOn.Never
-  };
-
-  // Create the language client and start the client.
-  const forceDebug = process.env['NG_DEBUG'] === 'true';
-  const client =
-      new lsp.LanguageClient('Angular Language Service', serverOptions, clientOptions, forceDebug);
+  const client = new AngularLanguageClient(context);
 
   // Push the disposable to the context's subscriptions so that the
   // client can be deactivated on extension deactivation
   registerCommands(client, context);
-  context.subscriptions.push(client.start());
-
-  client.onDidChangeState((e: lsp.StateChangeEvent) => {
-    if (e.newState === lsp.State.Running) {
-      registerNotificationHandlers(client);
-    }
-  });
-}
-
-function registerNotificationHandlers(client: lsp.LanguageClient) {
-  client.onNotification(notification.ProjectLoadingStart, () => {
-    vscode.window.withProgress(
-        {
-          location: vscode.ProgressLocation.Window,
-          title: 'Initializing Angular language features',
-        },
-        () => new Promise<void>((resolve) => {
-          client.onNotification(notification.ProjectLoadingFinish, resolve);
-        }),
-    );
-  });
 
-  client.onNotification(notification.RunNgcc, async (params: notification.RunNgccParams) => {
-    const {configFilePath} = params;
-    try {
-      await withProgress(
-          {
-            location: vscode.ProgressLocation.Window,
-            title: `Running ngcc for project ${configFilePath}`,
-            cancellable: false,
-          },
-          (progress: vscode.Progress<string>) => {
-            return resolveAndRunNgcc(configFilePath, progress);
-          },
-      );
-      client.sendNotification(notification.NgccComplete, {
-        configFilePath,
-        success: true,
-      });
-    } catch (e) {
-      vscode.window.showWarningMessage(
-          `Failed to run ngcc. Ivy language service might not function correctly. Please see the log file for more information.`);
-      client.sendNotification(notification.NgccComplete, {
-        configFilePath,
-        success: false,
-        error: e.message,
-      });
-    }
+  // Restart the server on configuration change.
+  const disposable = vscode.workspace.onDidChangeConfiguration(async () => {
+    await client.stop();
+    await client.start();
```

Comment:
Is it worth displaying a message that the server is restarting? I think no, what do you think?

Reply:
yeah, I don't think we need to display a message, because when the server restarts, it'll actually show "Initializing Angular language features ...". I think it's a good enough signal to indicate the server has restarted.